lib/wasix/src/net/mod.rs

```diff
@@ -200,7 +200,7 @@ pub(crate) fn write_ip_port<M: MemorySize>(
     ip: IpAddr,
     port: u16,
 ) -> Result<(), Errno> {
-    let p = port.to_be_bytes();
+    let p = port.to_ne_bytes();
```

Comment:
This is not the libc struct, but an internal wasi structure, so choice little endian / big endian is ours. Considering WebAssembly is LittleEndian, convertion to le is preferable.
On the `wasix-libc` side, this structure is read and the libc struct for addr is filed, with the port value being converted from host endianness (so LittleEndian because we are in WebAssembly) to Network endian (so BigEndian).
I suspect most wasm code that might use this structure will not convert port and will just read as it is (so le might be the best choice).

Reply:
So we have checked that the Rust codebase also does not do conversions, so I think the way to go is to use `to_le_bytes()`.

Reply:
We can just do:

```suggestion
    let p = port.to_le_bytes();
```